commit 13367a7858b58778ee0ad3f8ffab6092f49f3eba
Author: Jamie Coldhill <wrxtasy@amnet.net.au>
Date:   Sun May 8 15:06:59 2016 +0800

    FIX: Properly cache if both audio and video are starving
    Triggering a resync when only audio is starving practically prevents re-caching

diff --git a/xbmc/cores/dvdplayer/DVDPlayer.cpp b/xbmc/cores/dvdplayer/DVDPlayer.cpp
index 4ffe33a..bb5f79c 100644
--- a/xbmc/cores/dvdplayer/DVDPlayer.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayer.cpp
@@ -1830,7 +1830,7 @@ void CDVDPlayer::HandlePlaySpeed()
     // due to i.e. discontinuities of pts the stream may have drifted away
     // from clock too far for audio to sync back.
     if (m_CurrentAudio.id >= 0 && m_CurrentAudio.inited &&
-        m_dvdPlayerAudio->IsStalled() && m_dvdPlayerAudio->GetLevel() == 0)
+        m_dvdPlayerAudio->IsStalled() && m_dvdPlayerVideo->GetLevel() > 50)
     {
       CLog::Log(LOGDEBUG,"CDVDPlayer::HandlePlaySpeed - audio stream stalled, triggering re-sync");
       TriggerResync();
